src/nativeComponents.ts

```diff
@@ -88,13 +88,21 @@ function readMetadata(elfPath: string) {
     );
   }
 
+  let platformApiVersion;
+  try {
+    platformApiVersion = findSection('app_apiversion').toString();
+  } catch (ex) {
+    platformApiVersion = 'N/A';
```

Comment:
Is it possible to set the `platformApiVersion` to `undefined` instead of `N/A`. At the moment it works but in the future, setting to an invalid string (`N/A` in our case) is error prone.

Reply:
I tried that approach at first, but I got into this error which I couldn't solve.

[17:49:16][error][build] x.bundle Error: Unknown bundle component tag: Invalid value undefined supplied to : (({ type: "device", family: string, platform: Array<string>, platformCApiVersion: string } & Partial<{ isNative: true }>) | { type: "companion" })/0: ({ type: "device", family: string, platform: Array<string>, platformCApiVersion: string } & Partial<{ isNative: true }>)/0: { type: "device", family: string, platform: Array<string>, platformCApiVersion: string }/platformCApiVersion: string

I think that once I defined a tag inside component tags it always searches for it.

Reply:
Any suggestion on how to fix this is appreciated.

Reply:
io-ts can be a little difficult sometimes. What you really want to express here I think is that `platformCApiVersion` can only be set if `isNative` is true. I think you can achieve that using this:

```
const ComponentBundleTag = t.taggedUnion('type', [
  t.intersection([
    t.type({
      type: t.literal('device'),
      family: t.string,
      platform: t.array(t.string),
    }),
    t.union([
      t.type({
        isNative: t.literal(true),
        platformCApiVersion: t.string,
      }),
      t.partial({
        isNative: t.union([t.literal(true), t.undefined]),
        platformCApiVersion: t.undefined
      })
    ]),
  ]),
  t.type({
    type: t.literal('companion'),
  }),
]);
type ComponentBundleTag = t.TypeOf<typeof ComponentBundleTag>;
```

Further down, you can conditionally include the property into the manifest (if set), using:

```
...(bundleInfo.platformCApiVersion && { platformCApiVersion: bundleInfo.platformCApiVersion }),
```

Reply:
This works! Thank you so much for your help!